pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -387,14 +387,15 @@ ssize_t sock_dtls_recv(sock_dtls_t *sock, sock_dtls_session_t *remote,
                        void *data, size_t max_len, uint32_t timeout)
 {
     xtimer_t timeout_timer;
+    int is_timed_out = 0;
```

Comment:
this should be `volatile int` because it is changed from the callback that is executed in interrupt context.

Reply:
Added `volatile`. I had to cast `is_timed_out` to `void *` when passing it as `arg` of the timer (see line 398). I'm not sure if that affects the effect of `volatile.

Reply:
It's fine like that. The volatile is only relevant for the variable value within the `sock_dtls_recv` context. The pointer itself doesn't need to be volatile as the address itself is not changed.